### PR TITLE
feat(thumbnail): Add multipage thumbnail support.

### PR DESCRIPTION
**Pull request by ERosendo**
This PR adds support generating multiple thumbnails from a PDF and resolves https://github.com/freelawproject/doctor/issues/126.

This new service returns a .zip file containing thumbnails for each page requested.

### CODE REVIEW

**Review by mlissner — CHANGES_REQUESTED** [review]
Looks good. Zips are kind of annoying to deal with, but Python makes them friendly enough. 

One thing you could do, just for "fun" is to guard against [zip bombs](https://www.unforgettable.dk/). This isn't really getting input from random strangers on the Internet, but we've got the code to guard against them over on CourtListener, so it's easy to copy over. Start here:

https://github.com/freelawproject/courtlistener/blob/818ff303981cd96480e6797d8ec666e82a369aad/cl/recap/tasks.py#L410-L413

I also considered briefly whether just using a marker between files would work better. You could return a binary blob shaped like:

```
binary code for thumb 1
FLP_MARKER
binary code for thumb 2
FLP_MARKER
binary code for thumb 3
...
FLP_MARKER
binary code for thumb n
```

Then consumers just do:


```
data = response.content
files = data.split("FLP_MARKER")
```

That *is* a bit easier, but, meh, zips are standard and I guess that forces everything into memory which might be annoying. Anyway, it's a thought, but probably not worth doing.

Rest of the comments are pretty minor.

File: doctor/views.py [hunk 1]
```diff
@@ -139,6 +141,35 @@ def make_png_thumbnail(request) -> HttpResponse:
...
+    """
+    form = ThumbnailForm(request.POST, request.FILES)
+    if not form.is_valid():
+        return HttpResponse("Failed validation", status=BAD_REQUEST)
+
+    directory = TemporaryDirectory()
+
+    tmp_file = NamedTemporaryFile(suffix=".pdf")
+    with open(tmp_file.name, "wb") as f:
+        f.write(form.cleaned_data["file"].read())
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply doctor/views.py]: You're right. 

Changed. Thanks.

File: README.md [hunk 2]
```diff
@@ -216,9 +216,10 @@ Keep in mind that this curl will also write the file to the current directory.
 Give a PDF and a range or pages, this endpoint will return a zip file containing thumbnails
 for each page requested.  For example if you want thumbnails for the first four pages:
 
-    curl 'http://localhost:5050/convert/pdf/thumbnails/?max_dimension=350&pages=%5B1%2C+2%2C+3%2C+4%5D' \
+    curl 'http://localhost:5050/convert/pdf/thumbnails/' \
      -X 'POST' \
      -F "file=@doctor/test_assets/vector-pdf.pdf" \
+     -F 'pages="[1,2,3,4]"'
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply README.md]: Yes, the `-F` command line option tells `Curl` to send HTML data to the server in `multipart/form-data` format. I will update the README file to include the `max_dimension` parameter.

**Comment by ERosendo** [comment]
Thank you! @mlissner I've applied the requested changes.

We can also use the [FileExtensionValidator](https://docs.djangoproject.com/en/4.1/ref/validators/#fileextensionvalidator) to limit the input of this new service. This class from `django.core.validators` checks the extension of the file that's been uploaded and raises a [ValidationError](https://docs.djangoproject.com/en/4.2/ref/exceptions/#django.core.exceptions.ValidationError) if the extension is not allowed. If We use this validator class, the `ThumbnailForm` will look like this:

```python
class ThumbnailForm(forms.Form):
  file = forms.FileField(
      label="document",
      required=True,
      validators=[FileExtensionValidator(["pdf"])],
  )
  max_dimension = forms.IntegerField(label="max-dimension", required=False)
  pages = forms.Field(label="pages", required=False)
```

Let me know what you think.

**Comment by mlissner** [comment]
Sure, might as well use the validator, why not.

**Comment by mlissner** [comment]
The rest looks good. If you want to do the validator, go for it. If you want to add the zip bomb protection, go for it. Whatever you do, please merge when you're happy.

**Review by mlissner — APPROVED** [review]

**Pull request merged**